src/protocols/user_directed_commissioning/UDCClientState.h

```diff
@@ -75,6 +75,15 @@ class UDCClientState
     uint16_t GetLongDiscriminator() const { return mLongDiscriminator; }
     void SetLongDiscriminator(uint16_t value) { mLongDiscriminator = value; }
 
+    uint16_t GetVendorId() const { return mVendorId; }
+    void SetVendorId(uint16_t value) { mVendorId = value; }
+
+    uint16_t GetProductId() const { return mProductId; }
+    void SetProductId(uint16_t value) { mProductId = value; }
+
+    const uint8_t * GetRotatingId() const { return mRotatingId; }
+    void SetRotatingId(const uint8_t * rotatingId) { memcpy(mRotatingId, rotatingId, sizeof(mRotatingId)); }
```

Comment:
What guarantees the incoming buffer is big enough?  Rotating ids do not have a fixed size attached to them.

The incoming buffer needs a length, and we need to store how much data we actually have.

And our getter needs to return the length (e.g. by returning a Span, not a pointer).

Reply:
we now have a rotating id length field tracked by this object. The span change has not been made, which would be an improvement but can be done in a future cleanup.